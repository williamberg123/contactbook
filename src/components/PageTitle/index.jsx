import StyledPageTitle from './styles';

export default function PageTitle() {
	return (
		<StyledPageTitle>
			<h1>Agenda</h1>
			<p>Seus contatos est√£o logo abaixo</p>
		</StyledPageTitle>
	);
}
