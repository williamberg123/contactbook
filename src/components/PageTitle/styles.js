import styled from 'styled-components';

const StyledPageTitle = styled.div`
	width: 100%;
	display: flex;
	flex-direction: column;
	align-items: center;
	padding: 30px 0;

	& h1 {
		font-size: 2.2rem;
		margin-bottom: 10px;
		color: #222;
	}

	& p {
		font-size: 1.7rem;
		color: #a3a3a3;
	}

	@media (max-width: 600px) {
		& h1 {
			font-size: 1.7rem;
		}

		& p {
			font-size: 1.3rem;
		}
	}
`;

export default StyledPageTitle;
