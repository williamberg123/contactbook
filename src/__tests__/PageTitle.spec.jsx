import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import PageTitle from '../components/PageTitle';

describe('<PageTitle />', () => {
	it('should render PageTitle correctly', () => {
		render(<PageTitle />);

		expect(screen.getByRole('heading', { name: /agenda/i })).toBeInTheDocument();
		expect(screen.getByText(/seus contatos estÃ£o logo abaixo/i)).toBeInTheDocument();
	});

	it('should match snapshot', () => {
		render(<PageTitle />);

		expect(screen.getByRole('heading', { name: /agenda/i }).parentElement).toMatchSnapshot();
	});
});
