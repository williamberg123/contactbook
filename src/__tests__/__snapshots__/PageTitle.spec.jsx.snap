// Vitest Snapshot v1

exports[`<PageTitle /> > should match snapshot 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 30px 0;
}

.c0 h1 {
  font-size: 2.2rem;
  margin-bottom: 10px;
  color: #222;
}

.c0 p {
  font-size: 1.7rem;
  color: #a3a3a3;
}

@media (max-width:600px) {
  .c0 h1 {
    font-size: 1.7rem;
  }

  .c0 p {
    font-size: 1.3rem;
  }
}

<div
  class="c0"
>
  <h1>
    Agenda
  </h1>
  <p>
    Seus contatos est√£o logo abaixo
  </p>
</div>
`;
